### PR TITLE
:window: :wrench: Cleanup StepMenu properties

### PR DESCRIPTION
**Pull request by timroes**
## What

Cleanups the properties of the StepMenu component. This component is used for all tabs and had some weird one-off behavior in it to show the status icon for the attempt tabs inside the job logs. Also it did even ignore the "running" state completely and just didn't show an icon.

This fix now shows properly the loading status icon for running attempts, as well as moves the one-off logic out and instead give a tab a generic `icon` property, instead of passing `status` (that was already modified in the parent component) and `isPartialSuccess` down separately to combine them into an icon again in the `Step` component.

### CODE REVIEW

File: airbyte-webapp/src/components/ui/StepsMenu/StepsMenu.tsx [hunk 1]
```diff
@@ -25,17 +24,17 @@ const Content = styled.div`
...
       {data.map((item, key) => (
         <Step
-          status={item.status}
-          isPartialSuccess={item.isPartialSuccess}
+          icon={item.icon}
           lightMode={lightMode}
           key={item.id}
           num={key + 1}
-          {...item}
+          name={item.name}
```
Comment (timroes) [inline airbyte-webapp/src/components/ui/StepsMenu/StepsMenu.tsx:36]: :information_source: We spreaded this.. but only for 2 properties, the rest was still explicitally passed on above ... :shrug:  I made them now all explicitally passed through.

File: airbyte-webapp/src/components/ui/StepsMenu/Step.tsx [hunk 2]
```diff
@@ -1,20 +1,14 @@
-import React, { useMemo } from "react";
+import React from "react";
 import styled from "styled-components";
 
-import StatusIcon from "components/StatusIcon";
-import { StatusIconStatus } from "components/StatusIcon/StatusIcon";
-
-import Status from "core/statuses";
```
Comment (edmundito) [inline airbyte-webapp/src/components/ui/StepsMenu/Step.tsx:7]: Makes sense to remove this from this component

**Review by edmundito — APPROVED** [review]
Did a very basic local test to confirm. Would it make sense to add a unit test or storybook to these?

**Comment by timroes** [comment]
I started going down cleaning this up a bit more, but found we're having some more issues with this component. Thus I'll merge this with the current review, and created https://github.com/airbytehq/airbyte/issues/17799 to track further refactoring.

**Pull request merged**